File: generators/swift/codegen/src/ast/__test__/doc-comment/snapshots/complete_doc_comment.swift
/// Authenticates a user with the service
///
/// This method attempts to authenticate a user using their credentials. It will check both local cache and remote service if necessary.
///
/// - Parameter username: The user's login name
/// - Parameter password: The user's password
/// - Parameter rememberMe: Whether to store credentials for future use
/// - Returns: An authentication token if successful
/// - Throws: AuthenticationError if credentials are invalid
/// - Throws: NetworkError if unable to reach the service

File: generators/rust/codegen/src/ast/__test__/doc-comment/snapshots/complete_doc_comment.rs
/// Authenticates a user with the service
///
/// This method validates user credentials against the authentication service and returns a token if successful.
///
/// # Arguments
///
/// * `username` - The user's login name
/// * `password` - The user's password
/// * `options` - Additional authentication options
///
/// # Returns
///
/// Authentication token if successful
///
/// # Examples
///
/// ```rust
/// let client = AuthClient::new();
/// let token = client.authenticate("user", "pass", None).await?;
/// ```
